### PR TITLE
Refactor java class of domain client package to kotilin

### LINKED ISSUE #1529: Migrate Java models to Kotlin Data classes of core:models
**Summary:**
`core:models `have most of data holder classes written in java, We need to upgrade them in kotlin data models

**Steps to reproduce:**

How can we reproduce again the issue?

**Expected behavior:**

What did you expect the app to do?

**Observed behavior:**

What did you see instead?  Describe your issue in detail here.

**Device and Android version:**

What make and model device (e.g., Samsung Galaxy S3) did you encounter this on?  What Android
version (e.g., Android 4.0 Ice Cream Sandwich or Android 6.0 Marshmallow) are you running?  Is it
 the stock
version from the manufacturer or a custom ROM?

**Screenshots:**

Can be created by pressing the Volume Down and Power Button at the same time on Android 4.0 and higher.

### PR DESCRIPTION
**Pull request by NiranjanNlc**
## Issue Fix
Fixes #1529 partially 

## Screenshots

No change in UI 

## Description
converted the java classes of the domain client package and clientmapper.java classes to kotilin

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/ClientDetailsMapper.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        val clientDetails = com.mifos.mobilewallet.model.domain.client.Client()
+        if (client != null) {
+            clientDetails.name = client.displayName
+            clientDetails.clientId = client.id.toLong()
+            clientDetails.externalId = client.externalId
+            clientDetails.mobileNo = client.mobileNo
+        }
+        return clientDetails
+    } //    public ClientPayload transformClientPayload(NewClient client) {
+    //
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/ClientDetailsMapper.kt]: Remove this comment

File: core/model/src/main/java/com/mifos/mobilewallet/model/domain/client/Client.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package com.mifos.mobilewallet.model.domain.client
+
+import android.os.Parcel
+import android.os.Parcelable
+
+data class Client(
```
Comment (therajanmaurya) [inline core/model/src/main/java/com/mifos/mobilewallet/model/domain/client/Client.kt:8]: Please implement using `Parcelable` annotation. This is not optimized code using convert java to kotlin.

File: core/model/src/main/java/com/mifos/mobilewallet/model/domain/client/NewClient.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package com.mifos.mobilewallet.model.domain.client
+
+import com.mifos.mobilewallet.model.utils.DateHelper
+
+class NewClient(
```
Comment (therajanmaurya) [inline core/model/src/main/java/com/mifos/mobilewallet/model/domain/client/NewClient.kt]: I mentioned in the ticket, convert into data classes not classes.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/ClientDetailsMapper.kt (hunk 1 above)
Comment (NiranjanNlc) [reply core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/ClientDetailsMapper.kt]: done

**Pull request closed without merging**